index.js

```diff
@@ -36,33 +38,65 @@ var gulpSass = function gulpSass(options) {
       opts.omitSourceMapUrl = true;
     }
 
-    callback = function(error, obj) {
-      if (error) {
-        return cb(new gutil.PluginError(
-          PLUGIN_NAME, error.message + ' ' + gutil.colors.cyan('line ' + error.line) + ' in ' + gutil.colors.magenta(error.file)
-        ));
-      }
+    //////////////////////////////
+    // Handles returning the file to the stream
+    //////////////////////////////
+    filePush = function filePush(sassObj) {
       // Build Source Maps!
-      if (obj.map) {
-        applySourceMap(file, JSON.parse(obj.map.toString()));
+      if (sassObj.map) {
+        applySourceMap(file, JSON.parse(sassObj.map.toString()));
       }
 
-      file.contents = obj.css;
+      file.contents = sassObj.css;
       file.path = gutil.replaceExtension(file.path, '.css');
 
       cb(null, file);
     };
 
-    sass.render(opts, callback);
+    if (sync !== true) {
+      //////////////////////////////
+      // Async Sass render
+      //////////////////////////////
+      callback = function(error, obj) {
+        if (error) {
+          return cb(new gutil.PluginError(
+            PLUGIN_NAME, error.message + ' ' + gutil.colors.cyan('line ' + error.line) + ' in ' + gutil.colors.magenta(error.file)
+          ));
+        }
+        filePush(obj);
+      };
+
+      sass.render(opts, callback);
+    }
+    else {
+      //////////////////////////////
+      // Sync Sass render
+      //////////////////////////////
+      try {
+        result = sass.renderSync(opts);
+
+        filePush(result);
+      }
+      catch(error) {
+        return cb(new gutil.PluginError(
+          PLUGIN_NAME, error.message + ' ' + gutil.colors.cyan('line ' + error.line) + ' in ' + gutil.colors.magenta(error.file)
+        ));
```

Comment:
Need the colors, it's hard to distinguish things without them, plus they're the standard gulp colors. If a terminal doesn't support them, black and white.

As for the error message, do we really need to dry that out? Isn't that a bit overkill? 

> On Mar 28, 2015, at 1:21 PM, Vincent Prouillet <notifications@github.com> wrote:
> 
> In index.js:
> 
> > +
> > +      sass.render(opts, callback);
> > +    }
> > +    else {
> > +      //////////////////////////////
> > +      // Sync Sass render
> > +      //////////////////////////////
> > +      try {
> > +        result = sass.renderSync(opts);
> > +
> > +        filePush(result);
> > +      }
> > +      catch(error) {
> > +        return cb(new gutil.PluginError(
> > +          PLUGIN_NAME, error.message + ' ' + gutil.colors.cyan('line ' + error.line) + ' in ' + gutil.colors.magenta(error.file)
> > +        ));
> I'd extract the error formatting as well, also not a big fan of using too many colours since you never know what colour someone is using for their terminal bg
> 
> —
> Reply to this email directly or view it on GitHub.
>